mllib/src/main/scala/org/apache/spark/ml/recommendation/ALS.scala

```diff
@@ -374,8 +404,21 @@ class ALS(@Since("1.4.0") override val uid: String) extends Estimator[ALSModel]
   @Since("1.3.0")
   def setSeed(value: Long): this.type = set(seed, value)
 
+  /** @group expertSetParam */
+  @Since("2.0.0")
+  def setIntermediateRDDStorageLevel(value: String): this.type = {
+    set(intermediateRDDStorageLevel, value)
```

Comment:
The way params in the `ml` / DataFrame API work, is the validation happens in the Param definition, e.g. here: https://github.com/apache/spark/pull/12660/files#diff-be65dd1d6adc53138156641b610fcadaR165